evap/staff/forms.py

```diff
@@ -31,7 +31,27 @@ class ImportForm(forms.Form):
 
 
 class UserImportForm(forms.Form):
-    excel_file = forms.FileField(label=_("Excel file"))
+    excel_file = forms.FileField(label=_("Import from Excel file"), required=False)
+    course = forms.ModelChoiceField(Course.objects.all(), empty_label='<empty>', required=False, label=_("Copy from Course"))
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Here we Split the courses by Semester and create Supergroups for them. We also make sure to include an empty option.
+        choices = [('', '<empty>')]
+        for semester in Semester.objects.all():
+            course_choices = [(course.pk, course.name) for course in Course.objects.filter(semester=semester)]
+            if course_choices:
+                choices += [(semester.name, course_choices)]
+
+        # THis is the hacky part where we set the private property of the ModelChoiceField
+        self.fields['course']._choices = choices
```

Comment:
the comment is not helpful right now, it's just re-stating what the code does. either say *why* you needed to do the hack or remove it.

also, i've seen multiple stackoverflow posts where people wrote into `self.fields['foo'].choices`, without the underscore, does that not work?